meilenstein_02/analyse_Speichermedium/README.md

```diff
@@ -0,0 +1,105 @@
+# Initial Exploration Speichermedium
+
+<!-- Hier alles aufschreiben, was interessant erscheint! -->
+- In Welchen Größen werden die Daten von den Funktionen geliefert ?
+- In Welchen Größen möchte der DataClient Informationen erhalten ?
+- Wie sieht die Datenstruktur für die gespeicherten Daten aus ?
+- Werden die Daten als Dateien oder in einer Datenbank gespeichert ?
+- Was passiert bei einer Unterbechung zwischen Dataclient - Speichermedium und Funktion - Speichermedium ?
+- Werden noch weitere Funktionen als die Basics benötigt ?
+- Eventuell muss sich das Speichermedium noch um die Aufnahme der Configs für die Funktionen in den Dateien kümmern
+
+## Was sind die Ziele des Systems?
+<!-- Snow Cards können bei diesem Schritt helfen! -->
+- Es sollen die Daten von den drei Funktionen (sowie ihre eingestellte Config-Daten) gespeichert werden, jede Funktion speichert jeweils in einer eigenen Datei
+- Es ist möglich Dateien zu erstellen für verschiedene Datensätze der Funktionen von verschiedenen Versuchen
+- Es wird dem Dataclient eine Klasse mit Funktionen zur Verfügung gestellt, welche ihm ermöglichen nach Informationen, Dateien, als auch ganzen Datenordnern zu suchen
+- Das System ermöglicht es Datensätze auch wieder zu löschen
+
+## Was ist der erwartete Input?
+Es werden Datensätze von den Funktionen *Zoom Maps*, *Codecharts*, *Webcam basiertes Eyetracking* und DataClient erwartet. 
+
+## Was ist der erwartete Output?
+Es wird erwartet, dass der DataClient nach Dateien, Ordnern, als auch einzelnen Informatione fragt und diese ihm dann ausgegeben werden müssen.
+
+## Was sind Nomen Phrasen?
+<!-- Alle relevanten Sachen aufschreiben, später kann aussortiert werden! -->
+- Speicher/Schreiben
+- Lesen/Suchen
+- Löschen/Aendern
+- Sortieren
+
+## Liste der Klassen:
+<!-- Erstmal alle aufschreiben, dann auswählen! (Kriterien siehe Vorgehensweise) -->
+<!-- Warum sind die Klassen existent? Wenn das zu beantworten ist - u good! -->
+<!-- ausgewählte Klassen mit Link, andere einklammern und CRC-Karte löschen -->
+- [Suchen](crc-{Suchen}.md)
+- [Speichern](crc-{Speichern}.md)
+- [Aendern](crc-{Aendern}.md)
+- (Sortieren)
+
+---
+
+# Analyse - Speichermedium
+<!-- Hier Notizen zum Denkprozess! -->
+**Gedanken von Anfang bis Ende über das Speichermedium**
+1. Welche Aufgaben hat das Speichermedium/soll es zur Verfügung stellen?
+- ***Es muss zuerst die Art des Speichermediums entschieden werden***.. (Datenbank oder Datei) ..
+***Da dies noch in Besprechung ist, gehe ich aktuell von eine Datei aus.***
+- Neben dem **Als was** etwas abgespeichert werden soll gibt es noch andere Anforderungen
+- Es existiert eine abstrakte Oberklasse **SpeicherClient**, die in Verbindungen mit den Funktionen an sich steht
```

Comment:
```suggestion
- Es existiert eine Oberklasse **SpeicherClient**, die in Verbindungen mit den Funktionen an sich steht
```
Mir fällt gerade auf, `SpeicherClient` wäre ja gar nicht abstrakt, wird ja instanziiert, vereinigt ja in der Anwendung dann die Klassen unter sich
(nichts erbt von der Klasse)